src/model/card.ts

```diff
@@ -1,47 +1,177 @@
+
     }
 
+    /**
+     * Instantiates the cards and adds the updator function
+     * to the tagservice while also updating the dropdown section
+     */
     protected onInit(...args: any[]): void {
-        this.topic = args[0];
-        this. body = args[1];
+        this.topic = args[0][0];
+        this.body = args[0][1];
+        this.tags = [];
+        
+        //Initializing the containers for select and tagsection before
+        //creating the dom.
+        
+        this._dropDown = document.createElement('select');
+        this._tagSection = document.createElement('div');
+        this._tagSection.className = "tagsection";
+
+        //Function for adding new tags to card's drop down menu
+        this.tagService.addAgent(() => {
+            this._dropDown.options.length = 0;
+            for (const t of this.tagService.tags) {
+                let option: HTMLOptionElement = document.createElement('option');
+                option.text = t.name;
+                this._dropDown.add(option);
+            }
+        });
+        this.setTag(args[0][2]);
+        
+        //Make sure the tags are updated to latest
+        this.tagService.update();
     }
 
+    /**
+     * Generates a DOM model for current card. It follows this templte:
+     * <div class="card">
+     *  <span class="sec-topic">...</>
+     *  <span class="sec-body">...</>
+     *  <button class="tag">...</>
+     *  <div class="tag-section">...</>
+     *  <button class="rembutton">...</>
+     * </>
+     */
     protected domCreator() {
         const container = document.createElement('div');
         container.className = 'cards';
 
         const displayTopic = document.createElement('span');
+        container.className = 'sec-topic';
         displayTopic.textContent = this.topic;
-        
+
         const displayBody = document.createElement('span');
+        container.className = 'sec-body';
         displayBody.textContent = this.body;
 
+        const addTagButton = document.createElement('button');
+        addTagButton.textContent = 'AddTag';
+        addTagButton.className = "addtagbutton";
+        addTagButton.onclick = () => this.addTag();
+
         const removeButton = document.createElement('button');
         removeButton.textContent = 'Delete';
         removeButton.className = "rembutton";
-
         removeButton.onclick = () => {
             container.parentNode.removeChild(container);
             this.cardService.removeCard(this);
         };
 
         container.appendChild(displayTopic);
         container.appendChild(displayBody);
+        container.appendChild(this._dropDown);
+        container.appendChild(addTagButton);
+        container.appendChild(this._tagSection);
         container.appendChild(removeButton);
 
         this._dom = container;
     }
+
+    public resetDom(){
+    }
+
+    /**  
+     * By calling this, it will associate a specific tag with
+     * this object.
+     */
+    private addTag() {
```

Comment:
what is the difference of this and setTag?! and why you have repeated yourself exactly two lines after?
As a programmer, do you really think you should repeat your code?!